client/public/static/js/utils.js

```diff
@@ -238,23 +277,67 @@ const utils = (() => {
    * Handles post-login cookie verification and debugging.
    * 
    */
-  async function cookieWorkaround() {
+  /**
+
+* Handles post-login and post-verification cookie handling.
+
+* Verifies the session and fetches the user's profile to render UI dynamically.
+
+*/
+
+async function cookieWorkaround() {
+
+  try {
+
     const response = await fetch(PROFILE_URL, {
-      method: GET_ACTION,
-      credentials: 'include',
+
+      method: 'GET',
+
+      credentials: 'include', // Ensures cookies are sent with the request
+
       headers: {
-        'Content-Type': DEFAULT_DATA_TYPE,
+
+        'Content-Type': 'application/json',
+
       },
-    });
 
+    });
+ 
     if (!response.ok) {
-      throw new Error(FAILED_TO_GET_USER_PROFILE);
-    }
 
+      throw new Error('Failed to fetch user profile. Please log in again.');
```

Comment:
Can you make this string a constant that is pulled from constants.js like in other examples?

Reply:
Replace this string with your newly created constant and then should be good to go!